synapse/cores/common.py

```diff
@@ -604,6 +614,24 @@ def _bumpTufoCache(self, tufo, prop, oldv, newv):
     def _onSetAxonUrl(self, url):
         self.axon = s_telepath.openurl(url)
 
+    def _onSetMods(self, mods):
+        for ctor,conf in mods:
+            # load modules as dyndeps which construct a module
+            # subclass with ctor(<core>,<conf>)
+            try:
+
+                oldm = self.coremods.pop(ctor,None)
+                if oldm != None:
+                    oldm.fini()
+
+                modu = s_dyndeps.tryDynFunc(ctor, self, conf)
```

Comment:
Should we try to load this dep before unloading the old one? Is it OK that you can end up with an unloaded old module without loading a new module?